components/locale/src/extensions/unicode/key.rs

```diff
@@ -0,0 +1,87 @@
+use std::str::FromStr;
+
```

Comment:
Are transform/key and key essentially the same? Could they be combined somehow?

For example most of the boilerplate is the same (Display, FromStr, ...). I am already seeing some effects of code duplication, eg. use of KEY_LENGTH here and use of magic number for the same in transform/key.

If we notice heavy divergence in the future we can split them up again.

Reply:
The issue with deduplicating structs, even if they have the same code, is that to have complete docs I need docs on `unicode::Key` and `transform::Key`. There may be a way to extract the code maybe via macro or some other way and keep the docs sane, but I haven't found it, and would prefer not to overoptimize before landing.

I hope such optimizations can be done in process of work with the API.

Reply:
It can be done in follow up PR - I don't find this to be optimization as duplicate code elimination.

In C++/Java I would have common Impl code, and just simple wrappers in transform/key and key files.